runtime/memprof.c

```diff
@@ -785,30 +819,15 @@ CAMLprim value caml_memprof_set(value lv, value szv,
   trackst.entries = NULL;
   trackst.alloc_len = 0;
 
-  lambda = l;
-  if (l > 0) {
-    one_log1m_lambda = l == 1 ? 0 : 1/caml_log1p(-l);
-    next_mt_generate_geom = mt_generate_geom();
-  }
-
+  lambda = 0;
   caml_memprof_renew_minor_sample();
```

Comment:
You don't need to renew the sample anymore, right?

Reply:
It is actually necessary to reset the trigger to `young_alloc_start`, in order not to enter `caml_memprof_track_young` when `lambda = 0`.

Reply:
(We could do that manually, but I prefer keeping the invariant that only `shift_sample` and `caml_memprof_track_alloc_shr` change the trigger.)